syntax = "proto3";

package vehicle_detection;

service VehicleDetection {
  rpc Detect (ImageRequest) returns (VehicleResponse) {}
  rpc UpdateModel (ModelRequest) returns (ModelResponse) {}
}

message ImageRequest {
  bytes image_data = 1;
}

message Vehicle {
  int32 x1 = 1;
  int32 y1 = 2;
  int32 x2 = 3;
  int32 y2 = 4;
  string detected_class = 5;  // Changed from class to detected_class
  float confidence = 6;
}

message VehicleResponse {
  repeated Vehicle vehicles = 1;
}

message ModelRequest {
  string model_url = 1;
}

message ModelResponse {
  bool success = 1;
  string message = 2;
}
